### PR TITLE
Removed Python 2 code relating to six library

### PR DESCRIPTION
**Pull request by cwdavies**
## Removals

- Python 2 code in `six.PY2` block
- using `import six`
- use of `six.moves`
- `skipUnless` for `six.PY3`
- `if sys.version_info` check in regulations3k app
- python_2_unicode_compatible import

## Changes

- `six.text_type` changed to `str`
- StringIO import from `six` to `io`
- `six.StringIO` -> `io.StringIO`
- `six.moves.builtins.open` -> `builtins.open`
- use `import urllib` instead of `six.moves.urllib`
- use `import html.parser` instead of `from six.moves import html_parser`

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: cfgov/paying_for_college/disclosures/scripts/update_ipeds.py [hunk 1]
```diff
@@ -143,12 +143,6 @@ def download_files():
 def read_csv(fpath, encoding='utf-8'):
     if not os.path.isfile(fpath):
         download_files()
-    if six.PY2:  # pragma: no cover
-        with open(fpath, 'r') as f:
-            reader = cdr(f, encoding=encoding)
-            data = [row for row in reader]
-            return reader.fieldnames, data
-    else:  # pragma: no cover
         with open(fpath, newline='', encoding=encoding) as f:
```
Comment (higs4281) [inline cfgov/paying_for_college/disclosures/scripts/update_ipeds.py]: I think the `with` block needs to be un-indented a level here.

File: cfgov/paying_for_college/models/disclosures.py [hunk 2]
```diff
@@ -15,13 +14,8 @@
...
 import requests
 
-
-if six.PY2:  # pragma: no cover
-    from unicodecsv import writer as csw  # pragma: no cover
-    from unicodecsv import DictReader as cdr  # noqa # pragma: no cover
-else:  # pragma: no cover
-    from csv import writer as csw  # pragma: no cover
-    from csv import DictReader as cdr  # noqa # pragma: no cover
+from csv import writer as csw  # pragma: no cover
```
Comment (higs4281) [inline cfgov/paying_for_college/models/disclosures.py]:
```suggestion
from csv import writer as csw
```

Comment (higs4281) [reply cfgov/paying_for_college/models/disclosures.py]: No longer need to wave off coverage here (and in the next line as well)

File: cfgov/v1/tests/models/test_resources.py [hunk 3]
```diff
@@ -9,8 +8,8 @@
 class TestUnicodeCompatibility(TestCase):
     def test_unicode_resource_title_unicode(self):
         resource = Resource(title=u'Unicod\xeb')
-        self.assertEqual(six.text_type(resource), u'Unicod\xeb')
-        self.assertIsInstance(six.text_type(resource), six.text_type)
+        self.assertEqual(str(resource), u'Unicod\xeb')
```
Comment (higs4281) [inline cfgov/v1/tests/models/test_resources.py]:
```suggestion
        self.assertEqual(str(resource), 'Unicod\xeb')
```

Comment (higs4281) [reply cfgov/v1/tests/models/test_resources.py]: We could drop the `u` prefix

File: cfgov/v1/tests/models/test_snippets.py [hunk 4]
```diff
@@ -15,8 +14,8 @@
 class TestUnicodeCompatibility(TestCase):
     def test_unicode_contact_heading_unicode(self):
         contact = Contact(heading=u'Unicod\xeb')
-        self.assertEqual(six.text_type(contact), u'Unicod\xeb')
-        self.assertIsInstance(six.text_type(contact), six.text_type)
+        self.assertEqual(str(contact), u'Unicod\xeb')
```
Comment (higs4281) [inline cfgov/v1/tests/models/test_snippets.py]:
```suggestion
        self.assertEqual(str(contact), 'Unicod\xeb')
```

Comment (higs4281) [reply cfgov/v1/tests/models/test_snippets.py]: ditto.

File: cfgov/ask_cfpb/tests/models/test_pages.py [hunk 5]
```diff
@@ -37,11 +37,7 @@
 from v1.tests.wagtail_pages import helpers
 from v1.util.migrations import get_free_path, get_or_create_page
 
-
-if six.PY2:
-    import mock  # pragma: no cover
-else:
-    from unittest import mock  # pragma: no cover
+from unittest import mock  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/ask_cfpb/tests/models/test_pages.py]: Done

File: cfgov/core/utils.py [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import re
-from six import text_type as str
 from six.moves.urllib.parse import parse_qs, urlencode, urlparse
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/core/utils.py]: Done

File: cfgov/deployable_zipfile/tests/test_install_wheels.py [hunk 7]
```diff
@@ -25,9 +25,7 @@ def test_something(self, check_call):
 
         # We expect that only compatible wheels will be installed.
 
-        if six.PY2:  # pragma: no cover
-            expected_wheels = ['bar-1.0-py2-none-any.whl']
-        elif six.PY3:  # pragma: no cover
+        if six.PY3:  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/deployable_zipfile/tests/test_install_wheels.py]: Done

File: cfgov/paying_for_college/models/disclosures.py (hunk 2 above)
Comment (cwdavies) [reply cfgov/paying_for_college/models/disclosures.py]: Done

File: cfgov/v1/tests/models/test_resources.py (hunk 3 above)
Comment (cwdavies) [reply cfgov/v1/tests/models/test_resources.py]: Done

File: cfgov/v1/tests/models/test_snippets.py (hunk 4 above)
Comment (cwdavies) [reply cfgov/v1/tests/models/test_snippets.py]: Done

File: cfgov/paying_for_college/disclosures/scripts/load_programs.py [hunk 8]
```diff
@@ -98,23 +98,6 @@ def get_school(iped):
...
-                reader = cdr(f, encoding='windows-1252')
-                data = [row for row in reader]
-        except Exception:
-            data = [{}]
-    except Exception:
-        data = [{}]
-    return data
-
-
 def read_py3(filename):  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply cfgov/paying_for_college/disclosures/scripts/load_programs.py]: Done

**Pull request merged**